lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -6,53 +6,37 @@ class Deployment < KubernetesResource
     REQUIRED_ROLLOUT_TYPES = %w(maxUnavailable full none).freeze
     DEFAULT_REQUIRED_ROLLOUT = 'full'
 
-    def sync
-      raw_json, _err, st = kubectl.run("get", type, @name, "--output=json")
-      @found = st.success?
-
-      if @found
-        deployment_data = JSON.parse(raw_json)
-        @desired_replicas = deployment_data["spec"]["replicas"].to_i
-        @latest_rs = find_latest_rs(deployment_data)
-
-        @rollout_data = { "replicas" => 0 }.merge(deployment_data["status"]
-          .slice("replicas", "updatedReplicas", "availableReplicas", "unavailableReplicas"))
-        @status = @rollout_data.map { |state_replicas, num| "#{num} #{state_replicas.chop.pluralize(num)}" }.join(", ")
-
-        conditions = deployment_data.fetch("status", {}).fetch("conditions", [])
-        @progress_condition = conditions.find { |condition| condition['type'] == 'Progressing' }
-        @progress_deadline = deployment_data['spec']['progressDeadlineSeconds']
-        @max_unavailable = deployment_data.dig('spec', 'strategy', 'rollingUpdate', 'maxUnavailable')
-      else # reset
-        @latest_rs = nil
-        @rollout_data = { "replicas" => 0 }
-        @status = nil
-        @progress_condition = nil
-        @progress_deadline = @definition['spec']['progressDeadlineSeconds']
-        @desired_replicas = -1
-        @max_unavailable = @definition.dig('spec', 'strategy', 'rollingUpdate', 'maxUnavailable')
-      end
+    SYNC_DEPENDENCIES = %w(Pod ReplicaSet)
+    def sync(mediator)
+      super
+      @latest_rs = exists? ? find_latest_rs(mediator) : nil
+      @server_version = mediator.kubectl.server_version
```

Comment:
Its ugly but we could store this as a class level var to prevent every deployment from having to make the api call. Might not be worth the optimization.